### PR TITLE
Simplify `Nonoid.complex_generate` method

### PR DESCRIPTION
**Pull request by v-kolesnikov**
Hi!

It is a small enhancement of complex code construction and it brings a little perfomance improvement as well:

```ruby
require 'benchmark/ips'

Benchmark.ips do |x|
  x.report('original') do
    100_000.times do
      Nanoid.complex_generate(size: 21, alphabet: Nanoid::SAFE_ALPHABET)
    end
  end

  x.report('patch') do
    100_000.times do
      Nanoid.complex_generate_patch(size: 21, alphabet: Nanoid::SAFE_ALPHABET)
    end
  end

  x.compare!
end
```

Output:
```
Warming up --------------------------------------
            original     1.000  i/100ms
               patch     1.000  i/100ms
Calculating -------------------------------------
            original      0.967  (± 0.0%) i/s -      5.000  in   5.169805s
               patch      1.025  (± 0.0%) i/s -      6.000  in   5.854103s

Comparison:
               patch:        1.0 i/s
            original:        1.0 i/s - 1.06x  slower
```

Best regards!

### CODE REVIEW

File: lib/nanoid.rb [hunk 1]
```diff
@@ -25,21 +25,16 @@ def self.complex_generate(size:, alphabet:)
...
-          character = byte && alphabet[byte]
-          if character
-            id << character
-            throw :computed if id.size == size
-          end
-        end
+
+    loop do
+      bytes = random_bytes(size)
+      (0..step).each do |idx|
```
Comment (radeno) [inline lib/nanoid.rb]: Why did you change range from excluding last step to including last step?
I don't learn deeply why original JS implementation has "i < step" https://github.com/ai/nanoid/blob/master/format.js#L33 but there should be a reason.

**Comment by v-kolesnikov** [comment]
@radeno WDYT?

**Comment by radeno** [comment]
2 dots range vs 3 dots range

**Comment by v-kolesnikov** [comment]
Oh, it's my mistake, I'll fix it.

**Comment by v-kolesnikov** [comment]
It's fixed. @radeno

**Comment by v-kolesnikov** [comment]
Hm, I assume it was closed by my mistake (with "comment and close" button) and I didn't pay attention on it, I'm sorry. @radeno could you look on it again?

**Review by radeno — APPROVED** [review]
Looks good to me, cleaner solution than before
